pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -1429,15 +1429,17 @@ private void setConsumeEndTime(SegmentZKMetadata segmentZKMetadata, long now) {
     }
   }
 
-  private void fetchLatestStreamOffset() {
-    try (StreamMetadataProvider metadataProvider = _streamConsumerFactory
-        .createPartitionMetadataProvider(_clientId, _partitionGroupId)) {
-      _latestStreamOffsetAtStartupTime =
-          metadataProvider.fetchStreamPartitionOffset(OffsetCriteria.LARGEST_OFFSET_CRITERIA, /*maxWaitTimeMs*/5000);
+  public StreamPartitionMsgOffset fetchLatestStreamOffset() {
+    long maxWaitTimeMs = 5000;
```

Comment:
can we move this waitTime to be an argument `maxWaitTimeMillis`, since we seem to be calling it from different places and each may want to set their own wait time

Reply:
I'd prefer to leave as is for now since this is how it was working. It's a minimal change if/when anyone else needs it configurable.

Reply:
I am not worried about the minimal change afterwards. We have turned what was a private method in a class an changed it to a public method. At the minimum, the javadoc could indicate that there is 5s wait. At best, take a timeout parameter so that any other class that calls it does it knowingly.

What we don't want is another class calling this method without knowing that there could be a 5s delay and get surprised when there is

Reply:
fair, updated to make it a function parameter